File: nodejs/src/avenger/avenger.service.ts
import {BadRequestException, Injectable, NotFoundException} from '@nestjs/common';
import {AvengerRepository} from "./avenger.repository";
import {Avenger} from "./avenger.entity";

@Injectable()
export class AvengerService {
    constructor(private repository: AvengerRepository) {
    }

    async findAll(): Promise<Avenger[]> {
        return await this.repository.find();
    }

    async findOne(id: string): Promise<Avenger> {
        const existing = await this.repository.findOne(id);
        if (!existing) {
            throw new NotFoundException(`Avenger with the id:${id} doesn't exist.`);
        }
        return existing;
    }

    async create(avenger: Avenger): Promise<Avenger> {
        const existing = await this.repository.findOne({name: avenger.name});
        if (existing) {
            throw new BadRequestException(`Avenger with the name:${avenger.name} already exist.`);
        }
        return await this.repository.save(avenger);
    }

    async update(id: string, avenger: Avenger): Promise<Avenger> {
        const existing = await this.repository.findOne(id);
        if (!existing) {
            throw new NotFoundException(`Avenger with the id:${id} doesn't exist.`);
        }
        avenger.id = id;
        return await this.repository.save(avenger);
    }

    async delete(id: string): Promise<void> {
        const existing = await this.repository.findOne(id);
        if (!existing) {
            throw new NotFoundException(`Avenger with the id:${id} doesn't exist.`);
        }
        await this.repository.delete(id);
    }

}

File: java/src/main/java/io/egen/service/AvengerService.java
package io.egen.service;

import io.egen.entity.Avenger;
import io.egen.exception.BadRequestException;
import io.egen.exception.NotFoundException;
import io.egen.repository.AvengerRepository;
import org.springframework.stereotype.Component;

import java.util.List;
import java.util.Optional;

@Component
public class AvengerService {

    private AvengerRepository repository;

    public AvengerService(AvengerRepository repository) {
        this.repository = repository;
    }

    public List<Avenger> findAll() {
        return this.repository.findAll();
    }

    public Avenger findOne(String id) {
        Optional<Avenger> existing = this.repository.findById(id);
        if (!existing.isPresent()) {
            throw new NotFoundException("Avenger with the id:" + id + " doesn't exist.");
        }
        return existing.get();
    }

    public Avenger create(Avenger avenger) {
        Optional<Avenger> existing = this.repository.findByName(avenger.getName());
        if (existing.isPresent()) {
            throw new BadRequestException("Avenger with the name:" + avenger.getName() + " already exist.");
        }
        return this.repository.save(avenger);
    }

    public Avenger update(String id, Avenger avenger) {
        Optional<Avenger> existing = this.repository.findById(id);
        if (!existing.isPresent()) {
            throw new NotFoundException("Avenger with the id:" + id + " doesn't exist.");
        }
        avenger.setId(id);
        return this.repository.save(avenger);
    }

    public void delete(String id) {
        Optional<Avenger> existing = this.repository.findById(id);
        if (!existing.isPresent()) {
            throw new NotFoundException("Avenger with the id:" + id + " doesn't exist.");
        }
        this.repository.deleteById(id);
    }

}